### ISSUE TITLE
Customer got an error about Virtual Product after setting the shipping address on Cart that contains Downloadable Product only

### ISSUE THREAD

**Issue by TomashKhamlai** [magento/magento2#26107]
### Preconditions (*)

1. No preconditions

### Steps to reproduce (*)

1.  Create a cart with a downloadable product and try to set the shipping address

### Expected result (*)

1. A proper message

### Actual result (*)

1. Message about Virtual Product
```graphql
{
  "errors": [
    {
      "message": "The Cart includes virtual product(s) only, so a shipping address is not used.",
      "category": "graphql-no-such-entity",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "setShippingAddressesOnCart"
      ]
    }
  ],
  "data": {
    "setShippingAddressesOnCart": null
  }
}
```

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @lenaorobei 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-29905` were created

**Issue Available**: @lenaorobei, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by sanjay-wagento** [comment]
#MM20IN

**Comment by sanjay-wagento** [comment]
@lenaorobei I have reviewed the issue and it seems this is not related to graphql only. The message is comming from Magento core module. It seems above message is not related to virtula product only but it is for all product that does not contain shipping address.

If you think this is valid issue then can you please tell me which type of mesage you want to display through out system for downloadable, virtual and gift card virtula type products?

**Comment by lenaorobei** [comment]
Hi @sanjay-wagento! Please have a look at the comment from @pmclain. This is the best option we have now - change the exception message.

**Comment by magento-engcom-team** [comment]
Hi @TomashKhamlai. Thank you for your report. 
The issue has been fixed in magento/magento2#28904 by @michalderlatka in 2.4-develop branch
Related commit(s):
 - [1091a3089a0cf0c4fec43998dc1b9d82101b720b](https://github.com/magento/magento2/commit/1091a3089a0cf0c4fec43998dc1b9d82101b720b)
 - [c8c4e06b23a41beb24c347829b41b623c14b70db](https://github.com/magento/magento2/commit/c8c4e06b23a41beb24c347829b41b623c14b70db)
 - [77bd6138370b66f733c6b51e1d0d62422402dd12](https://github.com/magento/magento2/commit/77bd6138370b66f733c6b51e1d0d62422402dd12)

The fix will be available with the upcoming 2.4.1 release.
